### PR TITLE
Add deep dive section explaining the device handling behaviour of ivy

### PR DESCRIPTION
**Pull request by bipinKrishnan**
Adds the deep dive section explaining different ways to control the device on which the operation is performed, the soft device mode in Ivy and using `ivy.DefaultDevice` to force the operations on a specific device.

### CODE REVIEW

**Review by vedpatwardhan — APPROVED** [review]
lgtm! Thanks for updating the docs @bipinKrishnan, we can get the PR merged based on what @sherry30 thinks 😄👍

File: docs/overview/deep_dive/devices.rst [hunk 1]
```diff
@@ -135,6 +147,118 @@ This makes it clear that these backend-specific functions are only enterred into
...
+.. code-block:: python
+
+    ivy.set_backend("torch")
+    ivy.set_soft_device_mode(False)
+    x = ivy.array([1], device="gpu:0")
+    y = ivy.array([34], device="gpu:0")
+    ivy.add(x, y)
+
+The code to handle all these cases are present inside `@handle_device_shifting`_ decorator, which is wrapped around
+all the functions except mixed and compositional functions in `ivy.functional.ivy`_ submodule. The decorator calls
```
Comment (sherry30) [inline docs/overview/deep_dive/devices.rst:227]: All the functions that accept at least 1 array

**Review by sherry30 — CHANGES_REQUESTED** [review]
Just 1 minor addition.
Looks good otherwise.

Thanks!

**Review by sherry30 — APPROVED** [review]
lgtm!

**Pull request merged**